lib/stringify.js

```diff
@@ -112,6 +120,12 @@ module.exports = function (object, opts) {
     var sort = typeof options.sort === 'function' ? options.sort : null;
     var allowDots = typeof options.allowDots === 'undefined' ? false : options.allowDots;
     var serializeDate = typeof options.serializeDate === 'function' ? options.serializeDate : defaults.serializeDate;
+    if (typeof options.format === 'undefined') {
+        options.format = defaults.format;
+    } else if (!(options.format in defaults.formatters)) {
+        throw new TypeError('Unknown format options provided. Valid values are RFC3986 and RFC1738.');
```

Comment:
We have to notify user if provided format will not work. So, reaction to the wrong argument provided should be straight and self-explanatory